app/components/WidgetDynamicContent.tsx

```diff
@@ -229,12 +231,32 @@ export default class WidgetDynamicContent extends Component {
                             log.warn('Cannot attach event, missing data. Event data is ', event);
                             return;
                         }
-                        $(container).find(event.selector).off(event.event);
-                        $(container)
-                            .find(event.selector)
-                            .on(event.event, e => {
-                                event.fn(e, widget, this.getToolbox());
-                            });
+
+                        if (!container) {
+                            log.warn('Cannot attach event, missing container element. Event data is ', event);
+                            return;
+                        }
+
+                        if (this.attachedEvents.current === null) {
+                            this.attachedEvents.current = {};
+                        }
+
+                        if (this.attachedEvents.current[`${event.selector}_${event}`]) {
```

Comment:
Can we be a bit more explicit with the stringification of `event`?

```suggestion
                        if (this.attachedEvents.current[`${event.selector}_${JSON.strinfigy(event)}`]) {
```

Reply:
We could avoid the stringification if instead of a single `Record<string, (event: Event) => void>` we had: ` Record<string, Record<object, (event: Event) => void>>` (first map for the selectors, second map for the event type -> listener mapping)

Reply:
Minor:

Can we extract the condition to a separate variable to reuse it later (e.g. `const previousEventListener`)?